src/main/java/de/tum/in/www1/artemis/repository/ComplaintRepository.java

```diff
@@ -23,7 +25,11 @@
 @Repository
 public interface ComplaintRepository extends JpaRepository<Complaint, Long> {
 
-    Optional<Complaint> findByResult_Id(Long resultId);
+    Optional<Complaint> findByResultId(Long resultId);
+
+    default Complaint findByResultIdElseThrow(Long resultId) {
+        return findByResultId(resultId).orElseThrow(EntityNotFoundException::new);
```

Comment:
We should give some information in the EntityNotFoundException
Please have a look at other examples and use the same pattern